src/compiler/checker.ts

```diff
@@ -6299,9 +6298,19 @@ namespace ts {
 
                 function includePrivateSymbol(symbol: Symbol) {
                     if (some(symbol.declarations, isParameterDeclaration)) return;
-                    Debug.assertIsDefined(deferredPrivates);
+                    Debug.assertIsDefined(deferredPrivatesStack[deferredPrivatesStack.length - 1]);
                     getUnusedName(unescapeLeadingUnderscores(symbol.escapedName), symbol); // Call to cache unique name for symbol
-                    deferredPrivates.set(getSymbolId(symbol), symbol);
+                    // Blanket moving (import) aliases into the root private context should work, since imports are not valid within namespaces
+                    // (so they must have been in the root to begin with if they were real imports) cjs `require` aliases (an upcoming feature)
+                    // will throw a wrench in this, since those may have been nested, but we'll need to synthesize them in the outer scope

```

Comment:
I mean, a copy of `jsDeclarationsImportAliasExposedWithinNamespace.ts` but with cjs `require`s instead of `import` and `module.exports` instead of `export` should do; i'll add it, but you'll still need to watch it (since it may do... nothing).

Reply:
I've added `jsDeclarationsImportAliasExposedWithinNamespaceCjs.ts` - it currently has errors and unexpected output (due to the nested `typedef`s not resolving correctly with cjs exports).

Reply:
Thanks!